packages/amplify-cli/src/amplify-exception-handler.ts

```diff
@@ -79,6 +79,21 @@ export const handleException = async (exception: unknown): Promise<void> => {
   process.exit(1);
 };
 
+/**
+ * Handle rejected promises that weren't caught or awaited anywhere in the code.
+ */
+export const handleUnhandledRejection = (reason: Error | $TSAny): void => {
+  if (reason instanceof Error) {
+    throw reason;
+  } else if (reason !== null && typeof reason === 'string') {
+    throw new Error(reason);
+  } else if (reason !== null && typeof reason === 'object') {
+    throw new Error(JSON.stringify(reason));
+  } else {
+    throw new Error('Unhandled promise rejection');
+  }
```

Comment:
Two things here.

`console.log(typeof 1)` prints `number`. 
I suggest to simplify it and use JSON.stringify if not error and not null. it should handle both objects and primitives.

Should we throw `Error` or `AmplifyError`?

Reply:
1. This is done to prevent double quotation when printing strings (most common way of rejecting promises). E.g.
```
console.log(JSON.stringify("Message"));  // ""Message""
console.log("Message");                  // "Message"
console.log(JSON.stringify(1));          // "1"
console.log(1);                          // 1
```

2. This gets wrapped in `AmplifyError` by `handleException` which wraps a generic 'Error` to an `UnknownFault` which is the right categorization since we don't really know what caused the error. This effort ensures to capture the promise's rejected message. I'll add a comment here to make it clear.

Reply:
makes sense, but we should still make sure that all primitives (i.e. number) are captured.

Reply:
Makes sense. Update to capture everything and added a case for numbers.